components/gitpod-protocol/src/protocol.ts

```diff
@@ -567,7 +567,14 @@ export interface VSCodeConfig {
     extensions?: string[];
 }
 
+export interface RepositoryCloneInformation {
+    url: string;
+    checkoutLocation?: string;
+}
+
 export interface WorkspaceConfig {
+    mainConfiguration?: string;
```

Comment:
I'd propose we move this in a `multiRepo` section to keep te root namespace a bit cleaner, e.g.
```YAML
multiRepo:
  root: https://...
  children:
    - url: https://...
      checkoutLocation: foobar/
```

I'm not set on the naming at all, just tried to find something that's a tad shorter compared to what we have now. Current naming works, too:
```YAML
multiRepo:
  mainConfiguration: https://...
  additionalConfiguration:
    - url: https://...
      checkoutLocation: foobar/
```

Reply:
As it turns out these are not really bi-directional relations. `mainConfiguration` is simply delegating to a configuration using a context URL and doesn't necessarily imply that the main configuration holds a corresponding `additionalRepositories` entry. The `additionalRepositories` on the other hand is a list of repos to clone. It is thereby mainly an alternative to e.g. submodules that is aware of Gitpod context semantics (e.g. branches) and doesn't require a persistent pointer in the repo itself. So it makes sense to clone an additional repo without having a 'mainConfiguration' pointer on the other side. There are btw no use-cases I can think of where I would have both elements as in your example (but I guess it was just to clarify the schema you had in mind?).

Reply:
> As it turns out these are not really bi-directional relations.

💯 agree. This would make the `root`/`children` naming unfavourable. I reckon we'd still want to move things into a `multiRepo` namespace, if just to keep the root of the `.gitpod.yml` clean(er).

Reply:
I don't see how this helps, as there is only ever one item used in a `.gitpod.yml` and also the `mainConfiguration` is just a delegation mechanism and not necessarily a `multi-repo` thing. Let's have a brief sync conversation about this later.

Reply:
In practice no one will have both entries - let's skip the joint parent category.